tinygrad/codegen/linearizer.py

```diff
@@ -95,10 +96,11 @@ def global_load(self, i:int, idxs:List[Node], acc=None, barrier:Optional[UOp]=No
     invalid_value = 0
     for idx, valid, rep_idx in zip(e_idxs, e_valids, iter_idxs(expand_vars)):
       this_const, idx, valid = (invalid_value, NumNode(0), NumNode(1)) if valid.max == 0 else (const, idx, valid)
-      key = f"{acc}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
+      key = f"{acc is not None}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
```

Comment:
I need to think about this line, changing this key is tricky.
Can you write a test for it? I have one example in `test_load_cache_const_bufs`

Reply:
if you have multiple accs how are you ensuring they uop different DEFINE_ACCs?

Reply:
each reduceop is associated with a disjoint set of output buffers. self.reduceops.index(acc) is a good idea.

Reply:
well, we don't have self.reduceops yet, and self.reduceops might end up being 2 dimensional for both parallel and sequence.

Reply:
I figured it was fine to use the same local buffer for sequential reduceops as long as they're the same shape, that obviously won't work for parallel reduceops lol

there was some weirdness with the key, I also think we should pass `reduceop` to `global_load` and I was playing with `...{self.reduceops.index(reduceop) if reduceop else ''}...`

but I also start each reduce with an empty `self.load_cache` so it didn't end up needing to be changed